core/src/main/java/com/radixdlt/p2p/transport/PeerServerBootstrap.java

```diff
@@ -160,15 +161,17 @@ public void start() {
                 capabilities,
                 mempoolRelayerMaxMessagePayloadSize));
 
-    serverBind =
-        serverBootstrap.bind(config.listenAddress(), config.listenPort()).syncUninterruptibly();
+    serverBootstrap.bind(config.listenAddress(), config.listenPort()).syncUninterruptibly();
+    this.serverBootstrap.set(serverBootstrap);
 
     log.info("P2P server started. Node URI is: {}", self);
   }
 
   public void stop() {
-    if (serverBind != null) {
-      serverBind.channel().close().syncUninterruptibly();
+    @Nullable var runningServerBootstrap = this.serverBootstrap.getAndSet(null);
+    if (runningServerBootstrap != null) {
+      runningServerBootstrap.config().childGroup().shutdownGracefully().syncUninterruptibly();
+      runningServerBootstrap.config().group().shutdownGracefully().syncUninterruptibly();
```

Comment:
_(note from author)_

This fixes the runaway Netty pool problem found at https://github.com/radixdlt/babylon-node/pull/550#issuecomment-1638293010